src/slices/releaseSlice.js

```diff
@@ -0,0 +1,71 @@
+import { createSlice } from "@reduxjs/toolkit";
+import Axios from "axios";
+
+// First, define the reducer and action creators via `createSlice`
+export const releaseSlice = createSlice({
+  name: "releases",
+  initialState: { pending: false, error: "", items: [] },
+  reducers: {
+    getPending(state, action) {
+      state.pending = true;
+    },
+    getSuccess(state, action) {
+      state.pending = false;
+      state.items = action.payload;
+    },
+    getError(state, action) {
+      state.pending = false;
+      state.error = action.payload;
+    },
+    putIsPublicPending: state => {
+      state.pending = true;
+    },
+    putIsPublicSuccess: (state, action) => {
+      state.pending = false;
+      const newValue = action.payload.data.isPublic;
+      state.items.find(obj => {
+        if (obj.id == action.payload.id) {
+          obj.isPublic = newValue;
+        }
+      });
+    },
+    putIsPublicError: (state, action) => {
+      state.pending = false;
+      state.error = action.payload;
+    }
+  }
+});
+
+// Destructure and export the plain action creators
+export const {
+  getPending,
+  getSuccess,
+  getError,
+  putIsPublicPending,
+  putIsPublicSuccess,
+  putIsPublicError
+} = releaseSlice.actions;
+
+// Thunk for fetching
+export const fetchReleases = () => async dispatch => {
+  dispatch(getPending());
+  Axios.get("/releases")
+    .then(res => {
+      dispatch(getSuccess(res.data));
+    })
+    .catch(err => dispatch(getError(err)));
+};
+
+// Thunk for updating isPublic
+export const updateIsPublic = (id, isPublic) => async dispatch => {
```

Comment:
E der noken spesiell grunn til å ha en så spesialisert funksjon? I staden for å ha en generell `UpdateRelease()`

Reply:
Problemet kan vel også løses med å send heile release-objekt hver gang vi gjør ei endring. Den superlille fordelen eg ser med å ha fleire funksjona e en ytelsesfordel.

Reply:
He det forresten noke å sei om he he `async` før dispatch? E hadde inntrykket om at thunk gjor det under panseret

Reply:
Ikke sekker på om du må ha det eller ikke.